src/erlpmd_sup.erl

```diff
@@ -38,12 +39,29 @@
 start_link(Ips,Port) when is_list(Ips) ->
 	supervisor:start_link({local, ?MODULE}, ?MODULE, [Ips,Port]).
 
+start_link(Ips, Port, StoreMod, StoreOpts) when is_list(Ips), is_atom(StoreMod), is_list(StoreOpts) ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, [Ips, Port, StoreMod, StoreOpts]).
+
 %% ===================================================================
 %% Supervisor callbacks
 %% ===================================================================
 
-init([Ips,Port]) when is_list (Ips) ->
-	ErlPMD = {erlpmd, {erlpmd, start_link, [[]]}, transient, 5000, worker, [erlpmd]},
-	Listeners = [{{ip, Ip}, {tcp_listener, start_link, [[Ip,Port]]}, transient, 5000, worker, [tcp_listener]} || Ip <- Ips],
+init([Ips,Port | Opts ]) when is_list (Ips) ->
+    [StoreMod, StoreOpts] =
+        case Opts of
+            [] ->
+                % TODO This is really unclean. No opts if you set it via appenv?
```

Comment:
This needs some further thought/discussion. There has to be a neater/clearer way to set this up.